### ISSUE TITLE
Disable automatic sorting of the slices in the sunburst

### ISSUE THREAD

**Issue by xanderwallace85** [plotly/plotly.js#4823]
Hi guys!

Would it be possible to add an option that disables the automatic sorting of the slices in the sunburst?
Thank you in advance!

**Comment by emmanuelle** [comment]
This was also asked for on the forum https://community.plotly.com/t/order-of-the-sectors-in-sunburst-plot/39182.

Also see #4728 which is related.

**Comment by thierryVergult** [comment]
+1 for this request so that plotly sunburst does not sort (apply magic) on the data provided. 

I'm new to the plotly code base, but will try to find the code where sorting happens.

**Comment by thierryVergult** [comment]
In ./traces/sunburst/calc.js, sorting happens at

```
// TODO add way to sort by height also?
hierarchy.sort(function(a, b) { return b.value - a.value; });
```

Making that line conditional was solving my use case (no sorting).

Will try to add a configuration option, in line with branchvalues, to make this configurable.

**Comment by thierryVergult** [comment]
Is there a preference for a name of the layout attribute to disable sorting on values? One option is to reuse +sort+, an attribute available for the pie chart. Another option is to make it sunburst specific, like sunburstnosort.

**Comment by archmoj** [comment]
> Is there a preference for a name of the layout attribute to disable sorting on values? One option is to reuse +sort+, an attribute available for the pie chart. Another option is to make it sunburst specific, like sunburstnosort.

I think you could simply reuse `sort` for that.
```js
sort: extendFlat({}, pieAttrs.sort, {
  dflt: false
})
```

**Comment by archmoj** [comment]
Please note the `dflt` for `sunburst` (and `treemap`) should be set to `true` as their default.

**Comment by archmoj** [comment]
Resolved by #5164.

**Comment by nicolaskruchten** [comment]
OK. Here is where Plotly Express converts between the input data frame and the structure which go.Sunburst requires. Any changes to how things are sorted would need to be made here. https://github.com/plotly/plotly.py/blob/master/packages/python/plotly/plotly/express/_core.py#L1543
